providers/rfc2136/rfc2136.go

```diff
@@ -201,6 +206,11 @@ func (r *RFC2136Provider) sendMessage(msg *dns.Msg) error {
 		msg.SetTsig(r.tsigKeyName, dns.HmacMD5, 300, time.Now().Unix())
 	}
 
+	// use TCP transport if message exceeds the UDP payload limit
```

Comment:
@prachidamle i moved the size check below the `msg.SetTsig()` because the TSIG record will actually increase the message size.